experiments/mlp_mcdropout.py

```diff
@@ -54,16 +55,18 @@
 
 # Following Gal 2016, we reset the weights at the beginning of each step.
 initial_weights = deepcopy(model.state_dict())
+stopping_criterion = LabellingBudgetStoppingCriterion(
+    active_dataset=al_dataset, labelling_budget=10
+)
 
-for step in range(100):
+while True:
     model.load_state_dict(initial_weights)
     train_loss = wrapper.train_on_dataset(
         al_dataset, optimizer=optimizer, batch_size=32, epoch=10, use_cuda=use_cuda
     )
     test_loss = wrapper.test_on_dataset(test_ds, batch_size=32, use_cuda=use_cuda)
 
     pprint(wrapper.get_metrics())
-    flag = al_loop.step()
-    if not flag:
-        # We are done labelling! stopping
+    al_loop.step()
+    if stopping_criterion.should_stop():
```

Comment:
Wondering if you wanna change these two lines to below lines  to account for general exhaustion of the dataset.  The scenario does not apply to this example but as a demonstration of how to account for general exhaustion of the dataset and if the stop criteria is a metric limit

```suggestion
    flag = al_loop.step()
    if stopping_criterion.should_stop() or flag:
```

Reply:
My thinking was that stopping criteria must be added to ALLoop which would check for both the criteria and exhaustion. It would require ALLoop to know about metrics which it can't do right now.

That would be a major breaking change, so we might want to have a new object instead and deprecate ALLoop :thinking: and then this new object would do the entire experiment?